logical_switch.go

```diff
@@ -331,6 +331,66 @@ func (odbi *ovndb) lsExtIdsDelImp(ls string, external_ids map[string]string) (*O
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
+// lsAuxAddImp is for adding keys/vals either to external_ids or other_config record
+func (odbi *ovndb) lsAuxAddImp(ls string, auxConf map[string]string, auxCol string) (*OvnCommand, error) {
+	if (auxCol != "external_ids") && (auxCol != "other_config") {
+		return nil, fmt.Errorf("%s is neither other_config nor external_ids", auxCol)
+	}
+	if len(auxConf) == 0 {
+		return nil, fmt.Errorf("%s map is nil or empty", auxCol)
+	}
+
+	mutateSet, err := libovsdb.NewOvsMap(auxConf)
+	if err != nil {
+		return nil, err
+	}
+
+	var operations []libovsdb.Operation
+	row := make(OVNRow)
+	row["name"] = ls
+
+	mutation := libovsdb.NewMutation(auxCol, opInsert, mutateSet)
```

Comment:
per previous discussion with @girishmg, our understanding (per RFC 7047) is that if the key already exists, its value will not be updated. a special handling of this situation will be required - do we read back the entire record, scan all the keys, replace old values with the new ones, add new K/Vs, insert the merged record? Or let user deal with it. Otherwise, the new values for the old keys will be silently ignored, if the above assumption is correct.

Reply:
If this is something related to RFC7047 (which I agree it is), should we add this functionality to libovsdb to handle it? That way, all present and future ovn (and non-ovn) objects can benefit from it. WDYT?

Reply:
+1 for libovsdb handling it. makes more sense than on the client side, IMHO.

Reply:
I'm working on an object-based API for libovsdb and I was considering maybe it would be a good idea to implement operation logic, so we could end up with something like this: https://gorm.io/docs/update.html#Update-single-column

Even before that, we could add it to https://github.com/eBay/libovsdb/pull/30

Reply:
If we want to overwrite existing keys, according to RFC7047, we will have to use "update" instead of "mutate". We don't have to "read back the entire record" because it is already in the local cache. We can't do this in libovsdb right now because libovsdb doesn't maintain any cache. However, I agree that the object-based API may take this into consideration, but I haven't reviewed it, so not really sure.

Reply:
so, if the consensus is that go-ovn *Imp() are not the best place to deal with this right now, let's leave them as-is and merge this PR?

Reply:
@hzhou8 @vtolstov @noah8713 bump ... any chance for a merge?

Reply:
@vgozer, I think @hzhou8 is right, we can leverage the cache in go-ovn to figure out if we need to 'mutate' or 'update'. In order to implement the desired functionality you really need to do that or else it would silently ignore existing keys.

Reply:
@hzhou8 agree, let's decide on the proper way - this PR was meant to be a quick fix, but I hit a bunch of other issues, had to work them around with ... ovn-nbctl exec from the go code (sigh). Now we can spend more time on this and DTRT :)